src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -29,12 +29,12 @@ namespace Microsoft.AspNetCore.Mvc.Razor.Internal
     /// </summary>
     public class DefaultRoslynCompilationService : ICompilationService
     {
+        private readonly IHostingEnvironment _hostingEnvironment;
         private readonly IFileProvider _fileProvider;
         private readonly Action<RoslynCompilationContext> _compilationCallback;
         private readonly CSharpParseOptions _parseOptions;
         private readonly CSharpCompilationOptions _compilationOptions;
         private readonly ILogger _logger;
-        private readonly DependencyContext _dependencyContext;
```

Comment:
Will we now load the DC for **every** individual compilation? Or are the metadata references cached indefinitely?